doc/sphinx/language/core/sections.rst

```diff
@@ -3,57 +3,30 @@
 Section mechanism
 -----------------
 
-Sections create local contexts which can be shared across multiple definitions.
+Sections create naming scopes that permit creating section-local definitions that can
+be used by other definitions in the section.  Definitions made
+with :cmd:`Variable`, :cmd:`Hypothesis`, :cmd:`Let`, :cmd:`Let Fixpoint` and
+:cmd:`Let CoFixpoint` (or the plural variants of the first two) within sections
```

Comment:
Note that the most generic term (used in the CIC chapter for instance) is "declaration" which includes both "definitions" (`Let`) and "assumptions" (`Variable`). You shouldn't use "definition" as the generic term because definition implies "has a body".